libafl/src/state/mod.rs

```diff
@@ -798,6 +798,28 @@ where
         Ok(())
     }
 
+    /// Recursively walk supplied corpus directories
+    pub fn walk_initial_inputs<F>(
+        &mut self,
+        in_dirs: &[PathBuf],
+        mut closure: F,
+    ) -> Result<(), Error>
+    where
+        F: FnMut(&PathBuf) -> Result<(), Error>,
+    {
+        self.canonicalize_input_dirs(in_dirs)?;
+        loop {
+            match self.next_file() {
+                Ok(path) => {
+                    closure(&path)?;
+                }
+                Err(Error::IteratorEnd(_, _)) => break,
+                Err(e) => return Err(e),
+            }
+        }
+        self.reset_initial_files_state();
```

Comment:
Isn't this the exact opposite of what we want? 
We want to add all initial files in order to then load them - not clear them(?)
Keep in mind most fuzzers can restart during input loading

Reply:
@domenukk we are popping them anyways. For libafl-fuzz, we want to copy the corpora first to the queue directory before loading them, and thus the path will be different already and we cannot re-use ``remaining_initial_files`` etc.

Reply:
@domenukk remaining initial files will be empty by the time the walking the corpus or  loading the corpus is done but ``do_no_reenter`` may still be populated and this is why we need to reset the state.

Reply:
I guess I don't have to fully get it - but still,  `reset_initial_files_state` shoudn't be in a `walk_initial_inputs` method, they are unrelated.

Reply:
@domenukk why not? I'm thinking from a usage perspective. what's the point of having files in ``do_not_reenter`` after we're done walking? we won't traverse those symlinks again in ``load_initial_inputs`` without resetting the state.

Reply:
Ok I'm not 100% convinced this is the _cleanest_ implementation, but it is the one we have :)